nservicebus/nhibernate/saga-concurrency.md

```diff
@@ -26,8 +27,9 @@ snippet:NHibernateConcurrencyRowVersion
 
 That property will be included by NHibernate in the `SELECT` and `UPDATE` SQL statements causing concurrency violation error to be raised in case of concurrent updates.
 
-NOTE: Marking a property with `RowVersion` **does not** disable the pessimistic locking optimization. To switch to pure optimistic concurrency adjust the locking strategy to `Read`.
+NOTE: Marking a property with `RowVersion` **does not** disable the pessimistic locking optimization. All it does is replacing the default optimistic concurrency validation that depends on values of all columns with one that is based on that single explicit version column. To switch to pure optimistic concurrency adjust the locking strategy to `Read`.
 
+In most cases where the saga data table is only ever accessed by the saga persister it is advisable to use an explicit version because the `UPDATE` SQL statement is much simpler and faster. The downside is that it does not detect concurrency violations if the data is updated by some external party that does not conform to the protocol (does not bump the version field when doing updates). If such an external modification is possible it is better to use the default optimistic concurrency validation strategy.
```

Comment:
>The downside is that it does not detect concurrency violations if the data is updated by some external party that does not conform to the protocol (does not bump the version field when doing updates).

The downside is that it does not detect concurrency violations if the data is updated by some external party that does not conform to the protocol i.e. does not bump the version field when doing updates.

>If such an external modification is possible it is better to use the default optimistic concurrency validation strategy.

If such an external modification is possible, it is better to use the default optimistic concurrency validation strategy.

Can you clarify the scenario where an external party is making modifications?